spring2/src/main/java/com/bates/spring/UserController.java

```diff
@@ -0,0 +1,73 @@
+package com.bates.spring;
+
+
+import com.bates.spring.dto.UserDto;
+import main.java.com.bates.spring.model.Users;
+import main.java.com.bates.spring.model.request.UserRequest;
+import main.java.com.bates.spring.model.response.response.UserResponse;
+import com.bates.spring.services.UserService;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.beans.BeanUtils;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("users")
+public class UserController {
+
+    private final UserService userservice;
+
+    public UserController(UserService userservice){
+        this.userservice = userservice;
+    }
+
+    @GetMapping(path = "/allUsers")
+    public List<UserResponse> getAllUsers(){
+        List<UserResponse> returnValue = userservice.getAllUsers();
+        return returnValue;
+    }
+
+    @GetMapping
+    public List<UserDto> getUsers(@RequestParam(value = "page", defaultValue = "1") int page,
+                                  @RequestParam(value = "limit", defaultValue = "3") int limit){
+        List<UserDto> returnValue = userservice.getUsers(page, limit);
+        return returnValue;
+    }
+
+    @PostMapping
+    public UserResponse createUser(@RequestBody UserRequest userRequest){
+        UserDto userdto = new UserDto();
+        BeanUtils.copyProperties(userRequest, userdto);
+        UserDto createUser = userservice.createUser(userdto);
+        UserResponse userResponse = new UserResponse();
+        BeanUtils.copyProperties(createUser, userResponse);
+        return userResponse;
+    }
+
+    @GetMapping(path = "/{id}")
+    public UserResponse getUsers(@PathVariable Long id){
+        UserResponse returnValue = userservice.getUser(id);
+        System.out.println("Got here in controller");
+        return returnValue;
+    }
+    @GetMapping(path = "/email/{email}")
```

Comment:
dangerous and redundant. Get the email from the request body and use a public id in the path to identify the user. Nobody wants to see their private info in plain text in a copy-pasted URL